src/services/cron.js

```diff
@@ -1,18 +1,47 @@
 const cron = require("node-cron");
+const dotenv = require("dotenv")
+const userDB = require('../models/user');
+const paymentDB = require('../models/payment');
+const subscriptionDB = require('../models/subscription');
+const { db } = require('../models/user');
+const makePayment = require("./iyzico");
+
+dotenv.config({
+    path: "config.env",
+  });
+
+const price = process.env.PRICE;
+const interval = parseInt(process.env.INTERVAL);// 1 day of Interval for every succesfull subscriptions
 
 const scheduleSubscriptionPayment = () => {
   // Every day at 00:00:00
-  cron.schedule("0 0 0 * * *", () => {
+  cron.schedule("0 * * * * *", async() => {//checking every minute for test purpose!
     console.log("Running a task every day, date:", new Date());
+    const sub = await subscriptionDB.find();
 
-    /**
-     * TODO: Get records from Subscription collection
-     * We should have a logic to understand payment is needed or not.
-     * Then call makePayment service
-     * Note: It is ok to change Subscription schema if needed.
-     *
-     * */
+    if(sub){
+        for (let i = 0;sub[i].nextDate<=new Date();i++){//if sub[0].nextdate > sub[1].nextdate ==>not working!
+            const user = await userDB.findOne({userID : sub[i].userID});
+            const iyzicoResult = await makePayment(price , user);
```

Comment:
using Promise chaining(.then().then()) here would be meaningful to make the things faster.
So all records can make the payment in parallel.

Reply:
and also there is no try catch block to catch if those promises are rejected.
But if we use Promise chaining, it will not be needed anymore